netflix-roulette/src/pages/home/components/Catalog/components/GenresTabs/GenresTabs.tsx

```diff
@@ -1,23 +1,47 @@
 import * as React from "react";
+import { connect } from "react-redux";
 import { GENRES } from "@utils";
 import { Tabs, Tab } from "./GenresTabs.styles";
-import { useHomePageData } from "@data/HomePageDataProvider";
 import { GenreType } from "@utils/constants";
+import {
+  MoviesState,
+  FilterMovies,
+  FilterMoviesPayload,
+} from "@store/interfaces";
+import { filterMovies } from "@store/actions";
 
-export const GenresTabs = () => {
-  const { selectedGenre, setSelectedGenre } = useHomePageData();
+interface GenresTabsProps extends Partial<MoviesState> {
+  switchGenre: (payload: FilterMoviesPayload) => FilterMovies;
+}
 
-  return (
-    <Tabs>
-      {GENRES.map((genre: GenreType) => (
-        <Tab
-          key={genre}
-          isSelected={selectedGenre === genre}
-          onClick={() => setSelectedGenre(genre)}
-        >
-          {genre}
-        </Tab>
-      ))}
-    </Tabs>
-  );
+const GenresTabsComponent = ({
+  selectedGenre,
+  searchText,
+  switchGenre,
+}: GenresTabsProps) => (
+  <Tabs>
+    {GENRES.map((genre: GenreType) => (
+      <Tab
+        key={genre}
+        isSelected={selectedGenre === genre}
+        onClick={() => switchGenre({ selectedGenre: genre, searchText })}
+      >
+        {genre}
+      </Tab>
+    ))}
+  </Tabs>
+);
+
+const mapStateToProps = ({ moviesState }: { moviesState: MoviesState }) => ({
```

Comment:
Using `connect` API seems to be a bit old-fashioned these days, I believe, [`useSelector()` / `useDispatch()`](https://react-redux.js.org/api/hooks) hooks together with [reselect library](https://www.npmjs.com/package/reselect) would do a great job in your app.